src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -831,6 +915,50 @@ private static String readPropertiesContent(int beginLineNo, List<String> lines)
         return stringBuilder.toString();
     }
 
+    private static void validateProperties(Map<String, String> propertiesWithMissingDefaultTag,
+            List<String> unusedProperties) throws CheckstyleException {
+
+        if (!propertiesWithMissingDefaultTag.isEmpty()) {
+
+            final String propertiesList = propertiesWithMissingDefaultTag.entrySet().stream()
+                    .map(entry -> {
+                        return String.format(Locale.ROOT, "%s = (default)%s",
+                                entry.getKey(), entry.getValue());
+                    })
+                    .collect(Collectors.joining(", "));
+
+            final String message = String.format(Locale.ROOT,
+                    "Default properties must use the '(default)' tag."
+                    + " Properties missing the '(default)' tag: %s", propertiesList);
+            throw new CheckstyleException(message);
+        }
+        if (!unusedProperties.isEmpty()) {
+            final String message = String.format(Locale.ROOT,
+                    "All properties must be explicitly specified."
+                    + " Found unused properties: %s", unusedProperties);
+            throw new CheckstyleException(message);
+        }
+    }
+
+    private static void validateDefaultProperties(
+        Map<Object, Object> actualProperties,
+        Map<String, String> defaultProperties) throws CheckstyleException {
+        final Map<String, String> matchedProperties = actualProperties.entrySet().stream()
+                .filter(entry -> {
+                    return entry.getValue()
+                        .equals(defaultProperties.get(entry.getKey().toString()));
+                })
+                .collect(HashMap::new,
+                        (map, entry) -> {
+                        map.put(entry.getKey().toString(), entry.getValue().toString());
+                    }, HashMap::putAll);
+        final List<String> missingProperties = defaultProperties.keySet().stream()
+                .filter(propertyName -> !actualProperties.containsKey(propertyName))
+                .collect(Collectors.toUnmodifiableList());
+
+        validateProperties(matchedProperties, missingProperties);
```

Comment:
```suggestion
        validateProperties(matchedProperties(), missingProperties());
```

yes, i though the same. We need inline and give method. Field is just the same without giving deal dedication but using coupling. One thing only. This layer is dto and calles: `validateProperties(matchedProperties(), missingProperties());`

please consider SSOT applying separation of concern.

This layer is providing the props but the impl. detail in random and should live dedicated.

See the diff yourself.

Reply:
Then this layer seems `obsolet` as its just calling another method. Thats why we need clean code and bot boilerplate round.